profiling/src/internal/upscaling.rs

```diff
@@ -151,17 +151,20 @@ impl UpscalingRules {
         self.rules.is_empty()
     }
 
-    // TODO: Consider whether to use the internal Label here instead
-    pub fn upscale_values(
-        &self,
-        values: &mut [i64],
-        labels: &[pprof::Label],
-    ) -> anyhow::Result<()> {
+    pub fn upscale_values(&self, values: &mut [i64], labels: &[Label]) -> anyhow::Result<()> {
         if !self.is_empty() {
             // get bylabel rules first (if any)
             let mut group_of_rules = labels
                 .iter()
-                .filter_map(|label| self.get(&(StringId::new(label.key), StringId::new(label.str))))
+                .filter_map(|label| {
+                    self.get(&(
+                        label.get_key(),
+                        match label.get_value() {
+                            LabelValue::Str(str) => *str,
+                            LabelValue::Num { .. } => StringId::ZERO,
```

Comment:
Is this an error if this case is reached?

Reply:
I simply looked at `impl From<&Label> for pprof::Label` here.  https://github.com/DataDog/libdatadog/blob/0593cd71b018dcb53dcc37b82d7ab1306cbbf277/profiling/src/internal/label.rs#L59-L73

I don't think this was treated an error before this PR.